### PR TITLE
Null safety for hudson.model.Result

### PR DESCRIPTION
**Pull request by olivergondza**

### CODE REVIEW

File: core/src/main/java/hudson/model/Result.java [hunk 1]
```diff
@@ -49,42 +54,42 @@
     /**
      * The build had no errors.
      */
-    public static final Result SUCCESS = new Result("SUCCESS",BallColor.BLUE,0,true);
+    public static final @Nonnull Result SUCCESS = new Result("SUCCESS",BallColor.BLUE,0,true);
```
Comment (KostyaSha) [inline core/src/main/java/hudson/model/Result.java]: if you already changing this lines, please add spaces after comas

Comment (olivergondza) [inline core/src/main/java/hudson/model/Result.java]: This is inconsistent all over the place. I prefer not to touch it unless I have to rewrite the expression significantly.

File: core/src/main/java/hudson/model/Result.java [hunk 2]
```diff
@@ -133,24 +138,23 @@ public boolean isCompleteBuild() {
...
         return name;
     }
 
-    public String toExportedObject() {
+    public @Nonnull String toExportedObject() {
         return name;
     }
     
-    public static Result fromString(String s) {
+    public static @Nonnull Result fromString(@Nonnull String s) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/Result.java]: I don't like the original behavior of the method, but no way back

File: core/src/main/java/hudson/model/Result.java [hunk 3]
```diff
@@ -49,42 +54,42 @@
...
-    public static final Result NOT_BUILT = new Result("NOT_BUILT",BallColor.NOTBUILT,3,false);
+    public static final @Nonnull Result NOT_BUILT = new Result("NOT_BUILT",BallColor.NOTBUILT,3,false);
     /**
      * The build was manually aborted.
      *
      * If you are catching {@link InterruptedException} and interpreting it as {@link #ABORTED},
      * you should check {@link Executor#abortResult()} instead (starting 1.417.)
      */
-    public static final Result ABORTED = new Result("ABORTED",BallColor.ABORTED,4,false);
+    public static final @Nonnull Result ABORTED = new Result("ABORTED",BallColor.ABORTED,4,false);
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/Result.java]: May be an overkill, because static analysis tools will determine it automatically (field is final)

**Comment by oleg-nenashev** [comment]
:+1: with minor comments

File: core/src/main/java/hudson/model/Result.java (hunk 1 above)
Comment (KostyaSha) [inline core/src/main/java/hudson/model/Result.java]: spaces adds readability

**Pull request merged**